crates/oxc_minifier/src/ast_passes/peephole_substitute_alternate_syntax.rs

```diff
@@ -327,6 +347,143 @@ impl<'a> PeepholeSubstituteAlternateSyntax {
             None
         }
     }
+
+    fn try_fold_new_expression(
+        &mut self,
+        new_expr: &mut NewExpression<'a>,
+        ctx: &mut TraverseCtx<'a>,
+    ) -> Option<Expression<'a>> {
+        // `new Object` -> `{}`
+        if new_expr.arguments.is_empty()
+            && new_expr.callee.is_global_reference_name("Object", ctx.symbols())
+        {
+            Some(ctx.ast.expression_object(new_expr.span, Vec::new_in(ctx.ast.allocator), None))
+        } else if new_expr.callee.is_global_reference_name("Array", ctx.symbols()) {
+            // `new Array` -> `[]`
+            if new_expr.arguments.is_empty() {
+                Some(self.empty_array_literal(ctx))
+            } else if new_expr.arguments.len() == 1 {
+                let Some(arg) = new_expr.arguments[0].as_expression() else { return None };
```

Comment:
```suggestion
                let Some(arg) = new_expr.arguments.get_mut(0).and_then(|a| a.as_expression_mut())
                else {
                    return None;
                };
```
Use the `_mut()` method, which can finally return `&mut Expression`.

Reply:
In fact, the `get_mut` function may not be necessary in this scenario. However, we can utilize it to prevent panic in this particular situation.

Reply:
I like the `get_mut` here 👍